exporter/tanzuobservabilityexporter/metrics_test.go

```diff
@@ -197,6 +196,285 @@ func TestGaugeConsumerMissingValue(t *testing.T) {
+			Name:  malformedHistogramMetricName,
+			Value: float64(malformedCount),
+		})
+	assert.Contains(
+		t,
+		sender.metrics,
+		tobsMetric{
+			Name:  noAggregationTemporalityMetricName,
+			Value: float64(noAggregationTemporalityCount),
+			Tags:  typeIsHistogramTags,
+		})
+}
+
+func TestHistogramReportingError(t *testing.T) {
+	report := newHistogramReporting(zap.NewNop())
+	sender := &mockGaugeSender{errorOnSend: true}
+	var errs []error
+
+	report.Report(sender, &errs)
+
+	assert.NotEmpty(t, errs)
+}
+
+func TestCumulativeHistogramDataPointConsumer(t *testing.T) {
+	verifyCumulativeHistogramDataPointConsumer(t, false)
+}
+
+func TestCumulativeHistogramDataPointConsumerError(t *testing.T) {
+	verifyCumulativeHistogramDataPointConsumer(t, true)
+}
+
+func TestCumulativeHistogramDataPointConsumerLeInUse(t *testing.T) {
+	metric := newMetric("a.metric", pdata.MetricDataTypeHistogram)
+	histogramDataPoint := pdata.NewHistogramDataPoint()
+	histogramDataPoint.SetBucketCounts([]uint64{4, 12})
+	histogramDataPoint.SetExplicitBounds([]float64{10.0})
+	setTags(map[string]interface{}{"le": 8}, histogramDataPoint.Attributes())
+	sender := &mockGaugeSender{}
+	report := newHistogramReporting(zap.NewNop())
+	consumer := newCumulativeHistogramDataPointConsumer(sender)
+	var errs []error
+
+	consumer.Consume(metric, histogramDataPoint, &errs, report)
+
+	assert.Empty(t, errs)
+	assert.Empty(t, sender.metrics)
+	assert.Equal(t, int64(1), report.LeTagInUse())
+	assert.Equal(t, int64(0), report.Malformed())
+}
+
+func TestCumulativeHistogramDataPointConsumerMissingBuckets(t *testing.T) {
+	metric := newMetric("a.metric", pdata.MetricDataTypeHistogram)
+	histogramDataPoint := pdata.NewHistogramDataPoint()
+	sender := &mockGaugeSender{}
+	report := newHistogramReporting(zap.NewNop())
+	consumer := newCumulativeHistogramDataPointConsumer(sender)
+	var errs []error
+
+	consumer.Consume(metric, histogramDataPoint, &errs, report)
+
+	assert.Empty(t, errs)
+	assert.Empty(t, sender.metrics)
+	assert.Equal(t, int64(1), report.Malformed())
+	assert.Equal(t, int64(0), report.LeTagInUse())
+}
+
+// Creates a histogram metric with len(countAttributeForEachDataPoint)
+// datapoints. name is the name of the histogram metric; temporality
+// is the temporality of the histogram metric;
+// countAttributeForEachDataPoint contains the count attribute for each
+// data point.
+func newHistogramMetricWithDataPoints(
+	name string,
+	temporality pdata.MetricAggregationTemporality,
+	countAttributeForEachDataPoint []uint64,
+) pdata.Metric {
+	result := newMetric(name, pdata.MetricDataTypeHistogram)
+	histogram := result.Histogram()
+	histogram.SetAggregationTemporality(temporality)
+	histogram.DataPoints().EnsureCapacity(len(countAttributeForEachDataPoint))
+	for _, count := range countAttributeForEachDataPoint {
+		histogram.DataPoints().AppendEmpty().SetCount(count)
+	}
+	return result
+}
+
+func verifyCumulativeHistogramDataPointConsumer(t *testing.T, errorOnSend bool) {
```

Comment:
Can we pull this func into a separate happy-path test, which comes first in the Histogram tests? I'd like the edge cases and error handling tests to come after the happy-path test.

Reply:
Done.